runtime/near-vm-logic/src/gas_counter.rs

```diff
@@ -127,7 +121,15 @@ impl GasCounter {
             self.fast_counter.burnt_gas = new_burnt_gas;
             Ok(())
         } else {
-            Err(self.process_gas_limit(new_burnt_gas, new_burnt_gas + self.promises_gas).into())
+            // In the past `new_used_gas` would be computed using an implicit wrapping addition,
+            // which would then give an opportunity for the `assert` (now `debug_assert`) in the
+            // callee to fail, leading to a DoS of a node. A wrapping_add in this instance is
+            // actually fine, even if it gives the attacker full control of the value passed in
+            // here…
+            //
+            // [CONTINUATION IN THE NEXT COMMENT]
+            let new_used_gas = new_burnt_gas.wrapping_add(self.promises_gas);
```

Comment:
For API clarity reasons, I’d argue for making this into a `checked_add` still if we go for the protocol change route: having a function for which one parameter is "literally whatever" sounds wrong to me. That said it definitely would require a protocol change, so… that’s only if we decide to do the todo listed below I guess!